src/Fixer/ClassNotation/ClassAttributesSeparationFixer.php

```diff
@@ -166,13 +188,40 @@ protected function applyFix(SplFileInfo $file, Tokens $tokens)
      */
     protected function createConfigurationDefinition()
     {
-        $types = ['const', 'method', 'property'];
-
         return new FixerConfigurationResolver([
-            (new FixerOptionBuilder('elements', sprintf('List of classy elements; \'%s\'.', implode("', '", $types))))
+            (new FixerOptionBuilder('elements', 'Dictionary of `const|method|property` => `none|one` values.'))
                 ->setAllowedTypes(['array'])
-                ->setAllowedValues([new AllowedValueSubset($types)])
-                ->setDefault(['const', 'method', 'property'])
+                ->setAllowedValues([static function ($option) {
+                    foreach ($option as $type => $spacing) {
+                        if (!\in_array($type, self::SUPPORTED_TYPES, true)) {
```

Comment:
This is a breaking change: the previously allowed values won't be accepted anymore. The new implementation should detect the old format and:
* transform it into the new format;
* trigger a deprecation notice.